qa/rpc-tests/test_framework/authproxy.py

```diff
@@ -55,7 +55,7 @@
 
 class JSONRPCException(Exception):
     def __init__(self, rpc_error):
-        Exception.__init__(self)
+        Exception.__init__(self, rpc_error.get("message"))
```

Comment:
Why is this necessary?

Reply:
The problem with the bitcoin implementation is that it assumes that the passed-in exception has a `.error["message"]`. This probably only works for `JSONRPCException`. Since the test framework seems like it's trying to be somewhat generic, I felt it was a bad choice to tie `assert_raises_message` to this one type.

So instead I decided to get at the error string in a more generic way via `str(e)`. That necessitates that I pass in the rpc_error message to the base python `Exception` class. Passing the error message up to the base class has the benefit that anyone can now easily print out the exception string without knowing about the special internal structure that it has.